src/windows/export.py

```diff
@@ -903,13 +904,32 @@ def accept(self):
 
             progressstep = max(1 , round(( video_settings.get("end_frame") - video_settings.get("start_frame") ) / 1000))
             start_time_export = time.time()
+            seconds_run = 0
             start_frame_export = video_settings.get("start_frame")
             end_frame_export = video_settings.get("end_frame")
+            last_exported_time = time.time()
+            old_part = 0.0
+            new_part = 0.0
+            afterpoint = 1
+            # Precision of the progress bar
+            progress_format = "%4.1f%% "
             # Write each frame in the selected range
             for frame in range(video_settings.get("start_frame"), video_settings.get("end_frame") + 1):
                 # Update progress bar (emit signal to main window)
-                if (frame % progressstep) == 0:
-                    end_time_export = time.time()
+                end_time_export = time.time()
+                if ((frame % progressstep) == 0) or ((end_time_export - last_exported_time) > 1):
+                    new_part = (frame - start_frame_export) * 1.0  / (end_frame_export - start_frame_export)
+                    if ((new_part - old_part) > 0.0):
+                        afterpoint = math.ceil( -2.0 - math.log10( new_part - old_part ))
+                    else:
+                        afterpoint = 1
+                    if afterpoint < 1:
+                        afterpoint = 1
+                    if afterpoint > 5:
+                        afterpoint = 5
```

Comment:
These are the number of digits after the decimal point. Here they are set to a minimum of 1 and a maximum of 5. One was the previous default value 4.1f .
More than 5 doesn't make sense as in total this would be a higher precision than 7 digits. At 60 fps and 86400 seconds a day that would be 5,184,000 frames per day.
I have more of a problem with the maximum time between updates; I think 1 second is too low. 2 or 5 seconds might be better.

Reply:
I'm against `if (x < 20)` in the code when there is no comment why `20` was chosen here. Sometimes people like to make constants or variables that are much easier to modify later:
```
max_digits = 20; //max number of digits after the decimal point
if (x < max_digits) ...
```

Reply:
OK. I will make a new PR soon adding comments.
Today I might not have the time, soon.
Maybe also explain the calculation with the log10